server/web/session/redis/sess_redis.go

```diff
@@ -240,34 +251,46 @@ func (rp *Provider) SessionRead(ctx context.Context, sid string) (session.Store,
 
 // SessionExist check redis session exist by sid
 func (rp *Provider) SessionExist(ctx context.Context, sid string) (bool, error) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
 	c := rp.poollist
 
-	if existed, err := c.Exists(sid).Result(); err != nil || existed == 0 {
+	if existed, err := c.Exists(ctx, sid).Result(); err != nil || existed == 0 {
 		return false, err
 	}
 	return true, nil
 }
 
 // SessionRegenerate generate new sid for redis session
 func (rp *Provider) SessionRegenerate(ctx context.Context, oldsid, sid string) (session.Store, error) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
```

Comment:
I think we don't need this checkpoint because I think when the users call those methods with `context.Context` as the parameter, they should never pass nil.

So we don't need to be a babysitter.

Reply:
thanks for pointing this out! with the `nil` check removed, tests still pass.

but there's some code that calls `SessionRegenerate(nil, ...)` (same for `SessionRead`, `SessionExist`, `SessionDestroy` and `SessionInit`)

https://github.com/beego/beego/blob/219051dbc8cad7d92667dff7d69fb6e2f54c868b/server/web/session/session.go#L313C35-L313C47

does this need refactoring?

Reply:
change to `SessionRegenerate(context.Backgroud())` 😄